deployment-template/ec2/data-prepper-ec2-deployment-cfn.yaml

```diff
@@ -13,6 +13,9 @@ Parameters:
     AllowedPattern: "[a-z]+-[a-z]+-[0-9]+"
     Default: us-east-1
     ConstraintDescription: must be a valid AWS region (e.g. us-west-2)
+  AWSElasticsearchSubnetId:
+    Description: The subnet ID of the AWS Elasticsearch Service domain (Leave blank if the domain is not in a VPC)
+    Type: String
```

Comment:
Are there any constraints on this?

Reply:
From what I can tell no - the AWS console only indicates a limit of 256 characters. To test it out I made a subnet with weird characters named ` ######$$$$$$!!!!!!@@-----___++` which was created successfully. For that reason I didn't think it was worth adding a constraint.

Reply:
It is worth noting that CFN offers a Subnet parameter which populates a drop-down box of existing subnet IDs to choose from, however it can't be left blank. 

So if we were to create separate VPC/no-VPC templates we could use the Subnet parameter, but not in the combined template as it currently stands.

Reply:
yeah, not worth it. I'm good with it.